chia/consensus/block_body_validation.py

```diff
@@ -42,7 +42,7 @@ async def validate_block_body(
     height: uint32,
     npc_result: Optional[NPCResult],
     fork_point_with_peak: int,
-    get_block_generator: Callable[[BlockInfo, uint32], Awaitable[Optional[BlockGenerator]]],
+    get_block_generator: Callable[[BlockInfo, int, Dict[uint32, FullBlock]], Awaitable[Optional[BlockGenerator]]],
```

Comment:
I think we should bundle the `fork_height` and this height->block map into a class. We'll need to extend this with passing in the unspent coin set of the fork as well.

Also, I would worry about storing all the `FullNode` objects in memory. We're not likely to actually need all of that, and it can take up a lot of space

Reply:
what do you mean by "FullNode" objects?